### PR TITLE
Upconvert to rtb 2.6 (Some tests not working, need validation to support 2.6)

### PR DESCRIPTION
**Pull request by hhhjort**

### CODE REVIEW

File: endpoints/openrtb2/amp_auction.go [hunk 1]
```diff
@@ -178,6 +178,9 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		return
 	}
 
+	// upgrade to 2.6 here
+	openrtb_ext.ConvertUpTo26(reqWrapper)
```
Comment (bsardo) [inline endpoints/openrtb2/amp_auction.go:182]: Should this be moved into `parseAmpRequest` and be called right after the stored request is fetched but before we populate other fields like `req.test`, `req.imp[x].secure`, etc and perform validations?

File: endpoints/openrtb2/video_auction.go [hunk 2]
```diff
@@ -260,6 +260,8 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 	// all code after this line should use the bidReqWrapper instead of bidReq directly
 	bidReqWrapper := &openrtb_ext.RequestWrapper{BidRequest: bidReq}
 
+	openrtb_ext.ConvertUpTo26(bidReqWrapper)
```
Comment (bsardo) [inline endpoints/openrtb2/video_auction.go:263]: There are a few fields being set on the request like `id` and `test` prior to the upconvert but that should be ok.

File: endpoints/openrtb2/amp_auction_test.go [hunk 3]
```diff
@@ -765,15 +741,17 @@ func TestCCPAConsent(t *testing.T) {
...
+		/* if !assert.NotNil(t, result.Regs.Ext, test.description+":lastRequest.Regs.Ext") {
 			return
 		}
 		var re openrtb_ext.ExtRegs
 		err = jsonutil.UnmarshalValid(result.Regs.Ext, &re)
 		if !assert.NoError(t, err, test.description+":deserialize") {
 			return
 		}
-		assert.Equal(t, test.expectedRegExt, re, test.description)
+		assert.Equal(t, test.expectedRegExt, re, test.description) */
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply endpoints/openrtb2/amp_auction_test.go:752]: Yes, thanks.

File: endpoints/openrtb2/amp_auction_test.go [hunk 4]
```diff
@@ -765,15 +741,17 @@ func TestCCPAConsent(t *testing.T) {
...
+		assert.Equal(t, test.expectedRegExt.USPrivacy, *&result.Regs.USPrivacy, test.description+":USPrivacy")
+		assert.Equal(t, test.expectedRegExt.GDPR, *&result.Regs.GDPR, test.description+":GDPR")
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply endpoints/openrtb2/amp_auction_test.go:754]: I did not really consider that. The old code was the same, except looking to the unmarshalled ext from the result. I can try it without.

File: endpoints/openrtb2/amp_auction.go (hunk 1 above)
Comment (hhhjort) [reply endpoints/openrtb2/amp_auction.go:182]: I will move it up to just after we create the request wrapper. Seems to be a good place for it. The functions afterwards only seem to hit non-upconverted fields.

File: endpoints/openrtb2/video_auction.go (hunk 2 above)
Comment (hhhjort) [reply endpoints/openrtb2/video_auction.go:263]: Moving the conversion to the wrapper up would be a lot of work to make everything wrapper compatible. So as you said, it should work well here.

**Review by VeronikaSolovei9 — APPROVED** [review]

**Pull request merged**